cypress/e2e/signIn.cy.js

```diff
@@ -1,7 +1,31 @@
 /// <reference types="cypress" />
 
 describe('Sign In page', () => {
+  const randomNumber = Math.random().toString().slice(2, 5);
+  const username = `test${randomNumber}`;
+  const email = `${username}@gmail.com`;
+  const password = 'Testpass1!';
+
   it('should provide an ability to log in', () => {
+    cy.visit('https://conduit.mate.academy/');
+
+    cy.contains('Sign up').click();
+
+    cy.get(':nth-child(1) > .form-control').type(username);
+    cy.get(':nth-child(2) > .form-control').type(email);
+    cy.get(':nth-child(3) > .form-control').type(password);
+    cy.get('.btn').click();
+
+    cy.contains('Settings').click();
+
+    cy.contains('Or click here to logout.').click();
+
+    cy.contains('Sign in').click();
+
+    cy.get(':nth-child(1) > .form-control').type(email);
+    cy.get(':nth-child(2) > .form-control').type(password);
+    cy.get('.btn').click();
 
+    cy.get(':nth-child(4) > .nav-link').should('contain.text', username);
```

Comment:
You are checking if the username appears in the navigation bar to verify if the user has logged in. Although this works, a more reliable way would be to check if there is a session cookie or local storage item indicating that the user is logged in. Relying on UI elements for verifying login can be flaky and might lead to false negatives if there are changes in the UI.